docs: helix configuration for next-ls development

dbesserman opened the pull request:
As discussed on discord, here's the documentation about how to have helix interact with next-ls for development purposes (over TCP).

dbesserman committed:
docs: helix configuration for next-ls development

mhanberg commented:
Can you reword this so it's more about connecting via TCP? Also I don't think it needs the diff in the snippet

dbesserman commented:
Sure. Would you like me to add how to enable completions?

mhanberg commented:
Yes please!

dbesserman committed:
feat: helix configuration adjustments

Converts snippets from `diff` to `toml`.
The configuration for development has been changed to put more emphasis on the protocol rather than the tool.
Adds documentation to enable completions

mhanberg commented on _pages/docs/next-ls/editors.md:
````diff
@@ -181,6 +181,35 @@ language-servers = ["nextls"]
 command = "path/to/next-ls"
 args = ["--stdio=true"]
 ```
+### Development
````
```suggestion
### TCP
```

dbesserman replied:
I named this heading `Development` in order to maintain cohesion with the heading used in [the README.md file in the next-ls repo](https://github.com/elixir-tools/next-ls?tab=readme-ov-file#development). I understand how `TCP` can make more sense, but aren't you afraid some users will fail to make the connection between `Development` and `TCP` if `Development` isn't mentioned here? Maybe there's an other way than the heading the make the connection explicit.

mhanberg replied:
The perspective of this section of the docs is on editor configuration, not contributing. The word development doesn't necessarily imply developing Next LS. It could mean developing your application code.

mhanberg commented on _pages/docs/next-ls/editors.md:
````diff
@@ -181,6 +181,35 @@ language-servers = ["nextls"]
 ```
+### Development
+
+If you are running next-ls from the source code, you'll need to communicate with the server over TCP. [It is recommended to use netcat as a proxy.](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers)
+Assuming the next-ls server runs on port 9000, the LSP configuration should look as follows:
````
```suggestion
Helix supports connecting via TCP using `netcat`. https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers
```

mhanberg commented on _pages/docs/next-ls/editors.md:
````diff
@@ -181,6 +181,35 @@ language-servers = ["nextls"]
+args = ["127.0.0.1", "9000"]
+```
+
+### Enabling completions
````
```suggestion
### Initialization Options
```

mhanberg commented on _pages/docs/next-ls/editors.md:
```diff
@@ -181,6 +181,35 @@ language-servers = ["nextls"]
+
+### Enabling completions
+
+Completions need to be enabled explicitely in the Language Server configuration by passing intialization options under the `config` key.
```
```suggestion
Initialization options are configured with the `config` key.
```

dbesserman committed:
feat: apply suggestions from code review to helix documentation

Co-authored-by: Mitchell Hanberg <mitch@mitchellhanberg.com>
